### ISSUE TITLE
Linear regression compatible with gradient descent

### ISSUE THREAD

**Issue by TheEntertainerd** [TheEntertainerd/ml-from-scratch#4]
### Summary

Implement linear regression classes that work with the existing GradientDescent optimizer as well as testing

### Implementation Notes

- Create LinearRegressionOptimizable class implementing the Optimizable interface
- Add testing against pytorch 
- Create Manim animation showing

### Priority

High
